scan/lib/scan/runner.rb

```diff
@@ -135,37 +135,80 @@ def retryable_tests(input)
       return retryable_tests.uniq
     end
 
-    def handle_results(tests_exit_status)
-      if Scan.config[:disable_xcpretty]
-        unless tests_exit_status == 0
-          UI.test_failure!("Test execution failed. Exit status: #{tests_exit_status}")
-        end
-        return
+    def trainer_test_results
+      require "trainer"
+
+      results = {
+        number_of_tests_including_retries: 0,
+        number_of_failures_including_retries: 0,
+        number_of_tests: 0,
+        number_of_failures: 0
+      }
+
+      result_bundle_path = Scan.cache[:result_bundle_path]
+      output_path = Scan.config[:output_directory] || Dir.mktmpdir
+
+      UI.crash!("A -resultBundlePath is needed to parse the test results. This should not have happened. Please file an issue.") unless result_bundle_path
+
+      params = {
+        path: result_bundle_path,
+        output_path: output_path,
+        silent: true,
+        extension: "xml"
+      }
+
+      resulting_paths = Trainer::TestParser.auto_convert(params)
+      resulting_paths.each do |path, data|
+        results[:number_of_tests_including_retries] += data[:number_of_tests]
+        results[:number_of_failures_including_retries] += data[:number_of_failures]
+        results[:number_of_tests] += data[:number_of_tests_excluding_retries]
+        results[:number_of_failures] += data[:number_of_failures_excluding_retries]
       end
 
-      result = TestResultParser.new.parse_result(test_results)
```

Comment:
This is the whole point of this PR, right?🎯

Reply:
Ha, yup! That was the thing that relied junit test output from xcpretty 😛